src/components/calcite-color-picker/calcite-color-picker.tsx

```diff
@@ -606,8 +616,11 @@ export class CalciteColorPicker {
 
   /** Sets focus on the component. */
   @Method()
-  async setFocus(): Promise<void> {
-    focusElement(this.hexInputNode);
+  async setFocus(focusId?: "hex-input"): Promise<void> {
+    const elementToFocus =
+      focusId === "hex-input" ? this.hexInputNode : getFocusableElements(this.el.shadowRoot)[0];
```

Comment:
I think you can simplify this by checking which parts are hidden and focusing on the relevant refs since there's no user-provided content:

```
// pseudocode
if (focusIdHexInputAndNotHidden) 
  focusElement(hexInputNode)
else 
  focusElement(scopeNode)
```

Reply:
@jcfranco are you saying to not use `getFocusableElements`? Because it does that work already.

If we want to simplify, i can just get rid of the focusId argument and not allow a user to choose what to focus on.